keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1150 @@
+// - If the requested amount exceeds the maximum value in the available options, the request does not satisfy the policy,
+//   and the device cannot be allocated.
+type CapacitySharingPolicyDiscrete struct {
+    // Default specifies the default capacity to be used for a consumption request
+    // if no value is explicitly provided.
+    //
+    // +required
+    Default resource.Quantity
+
+    // Options defines a list of additional valid capacity values that can be requested.
+    // The Default must not be one of the Options.
+    //
+    // +optional
+    // +listType=atomic
+    Options []resource.Quantity
+}
+
+
+// CapacitySharingPolicyRange defines a valid range for consumable capacity values.
+//
+// - If the requested amount is less than Minimum, it is rounded up to the Minimum value.
+// - If the requested amount is between Minimum and Maximum, ChunkSize is set,
+//   and the amount does not align with the ChunkSize,
+//   it will be rounded up to the next value matching Minimum + (n * ChunkSize).
+// - If the requested or rounded amount exceeds Maximum (if set), the request does not satisfy the policy,
+//   and the device cannot be allocated.
+//
+// - If ChunkSize is not set, the requested amount is used as-is, provided it falls within the range.
+type CapacitySharingPolicyRange struct {
+    // Minimum specifies the minimum capacity allowed for a consumption request.
+    // This also acts as the default if no value is specified.
+    //
+    // +required
+    Minimum resource.Quantity
+
+    // Maximum defines the upper limit for capacity that can be requested.
+    //
+    // +optional
+    Maximum *resource.Quantity
+
+    // ChunkSize defines the step size between valid capacity amounts within the range.
+    // If set, requested amounts are rounded up to the nearest multiple of ChunkSize from the Minimum.
+    //
+    // +optional
+    ChunkSize *resource.Quantity
+}
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+```go
+
+type DeviceRequest struct {
+...
+   // CapacityRequests define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   CapacityRequests *CapacityRequirements
+}
+
+
+// CapacityRequirements defines the capacity requirements for a specific device request.
+type CapacityRequirements struct {
+    // Minimum defines the minimum amount of each device capacity required for the request.
+    //
+    // If the capacity has a sharing policy, this value is rounded up to the nearest valid amount
+    // according to that policy. The rounded value is used during scheduling to determine how much capacity to consume.
+    //
+    // If the quantity does not have a sharing policy, this value is used as an additional filtering
+    // condition against the available capacity on the device.
+    // This is semantically equivalent to a CEL selector with
+    // `device.quantity[<domain>].<name>.compareTo(<minimum quantity>) >= 0`.
```

Comment:
a little orthogonal, but would we be happy with someone using a CEL selector like that, or would we want them to guard it to make it safe using CEL optionals and a default of 0 to do this comparison? If so, let's get in the habit of putting CEL in docs we'd be happy with someone copy/pasting.

Also, is this `device.capacity` instead of `device.quantity`?

```
device.?capacity[<domain>].<name>.value.orValue(quantity("0")).compareTo(<minimum quantity>) >= 0
```